### PR TITLE
new crossfilter recipe with multiple outputs

### PR DESCRIPTION
**Pull request by nicolaskruchten**
This PR proposes a new, streamlined and easier to understand "generic crossfilter recipe" based off of multiple outputs instead of a harder-to-understand function generator.

I've basically unrolled most of the loops to make it clearer to Python beginners how the three figures inter-relate.

--- 

Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [x] I understand

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
Note: the output of this app is basically identical to the output of the previous recipe, so the GIF doesn't need to change :)

**Comment by nicolaskruchten** [comment]
image

File: dash_docs/tutorial/examples/crossfilter_recipe.py [hunk 1]
```diff
@@ -9,146 +9,91 @@
...
-
-
-
-# app.callback is a decorator which means that it takes a function
-# as its argument.
-# highlight is a function "generator": it's a function that returns function
-app.callback(
-    Output('g1', 'figure'),
+                'line': { 'width': 1, 'dash': 'dot', 'color': 'darkgrey' },
+                **selection_bounds
```
Comment (alexcjohnson) [inline dash_docs/tutorial/examples/crossfilter_recipe.py]: heh, never knew you could use `**` that way in a dict. I guess neither did Python2 😉

Comment (nicolaskruchten) [reply dash_docs/tutorial/examples/crossfilter_recipe.py]: hmmm... the old version did this? or maybe a slightly different flavour

**Comment by nicolaskruchten** [comment]
Huh so in py2 `{"a":2, **b}` doesn't work but `dict({"a":2}, **b)` does.

File: dash_docs/tutorial/examples/crossfilter_recipe.py (hunk 1 above)
Comment (alexcjohnson) [reply dash_docs/tutorial/examples/crossfilter_recipe.py]: looks like `dict(type='rect', line={...}, **selection_bounds)` would work, but not `**` inside `{}`

Comment (nicolaskruchten) [reply dash_docs/tutorial/examples/crossfilter_recipe.py]: jinx

**Pull request merged**